clients/drcachesim/simulator/simulator.cpp

```diff
@@ -87,6 +88,21 @@ simulator_t::init_knobs(unsigned int num_cores, uint64_t skip_refs, uint64_t war
     }
 }
 
+void
+simulator_t::create_v2p_from_file(std::string v2p_file_path)
+{
+    v2p_reader_t v2p_reader;
+    v2p_info_t v2p_info;
+    std::string error_str = v2p_reader.create_v2p_info_from_file(v2p_file_path, v2p_info);
+    if (!error_str.empty()) {
+        ERRMSG("Error: v2p_reader failed with: %s", error_str.c_str());
+        success_ = false;
```

Comment:
`success_` is for tool construction where the constructor has no return value.  As a separate utility function it seems better to return the error string.

Reply:
Done.